app/models/loan.rb

```diff
@@ -59,8 +59,13 @@ class Loan < ActiveRecord::Base
   belongs_to :representative, class_name: 'Person'
   belongs_to :organization_snapshot
 
-  has_many :project_steps, as: :project
-  has_many :project_logs, through: :project_steps
+  has_many :timeline_entries, as: :project
+  has_many :project_logs, through: :timeline_entries
+
+  # Do regular ruby select, to avoid issues with AR caching
+  def project_steps
+    timeline_entries.select{ |e| e.type == "ProjectStep" }
```

Comment:
For this project it doesn't matter.

On Tue, Sep 13, 2016 at 6:43 PM, Nathan Lee <notifications@github.com>
wrote:

> In app/models/loan.rb
> <https://github.com/sassafrastech/madeline_system/pull/136#discussion_r78658627>
> :
>
> > @@ -59,8 +59,13 @@ class Loan < ActiveRecord::Base
> >    belongs_to :representative, class_name: 'Person'
> >    belongs_to :organization_snapshot
> >
> > -  has_many :project_steps, as: :project
> > -  has_many :project_logs, through: :project_steps
> > +  has_many :timeline_entries, as: :project
> > +  has_many :project_logs, through: :timeline_entries
> > +
> > +  # Do regular ruby select, to avoid issues with AR caching
> > +  def project_steps
> > +    timeline_entries.select{ |e| e.type == "ProjectStep" }
>
> @hooverlunch <https://github.com/hooverlunch> where do we fall on string
> literals? https://github.com/bbatsov/ruby-style-guide#strings I prefer
> option B, but most projects I have worked on prefer A.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/sassafrastech/madeline_system/pull/136/files/d5be92abf5dd5774bf8353cd9021482a60387f44#r78658627>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAq3cIUxDIR5tKuJbeyeF1Oyevw4R4FYks5qpycbgaJpZM4J6NOi>
> .
>



-- 
Tom Smyth

Worker-Owner, Sassafras Tech Collective
Specializing in innovative, usable tech for social change
sassafras.coop *·* @sassafrastech